hoot-core/src/main/cpp/hoot/core/scoring/GraphComparator.cpp

```diff
@@ -72,13 +72,13 @@ GraphComparator::GraphComparator(OsmMapPtr map1, OsmMapPtr map2) :
       _mean(0.0),
       _ci(-1.0),
       _s(-1.0),
-      _maxGraphCost(0.),
-      _debugImages(false)
+      _debugImages(false),
+      _maxThreads(10)
 {
   _init();
 }
 
-cv::Mat GraphComparator::_calculateCostDistance(OsmMapPtr map, Coordinate c)
+cv::Mat GraphComparator::_calculateCostDistance(OsmMapPtr map, Coordinate c, double& maxGraphCost, const RandomPtr& random)
```

Comment:
To provide consistency the `Tgs::Random` object is passed in here, that way results are predictable for testing.